libs/datasets/sources/hhs_hospital_dataset.py

```diff
@@ -1,24 +1,119 @@
+import dataclasses
 from functools import lru_cache
+from libs import pipeline
+
 from covidactnow.datapublic.common_fields import CommonFields
+import pandas as pd
+from libs.datasets.sources import can_scraper_helpers as ccd_helpers
 from libs.datasets import data_source
 from libs.datasets.timeseries import MultiRegionDataset
 
 
-class HHSHospitalDataset(data_source.DataSource):
-    SOURCE_TYPE = "HHSHospital"
+# Early data is noisy due to lack of reporting, etc. so we just trim it off.
+DEFAULT_START_DATE = "2020-09-01"
+CUSTOM_START_DATES = {
+    "02": "2020-10-06",  # Alaska
+    "04": "2020-09-02",  # Arizona
+    "15": "2020-10-10",  # Hawaii
+    "16": "2020-10-17",  # Idaho
+    "19": "2020-09-05",  # Iowa
+    "21": "2020-10-15",  # Kentucky
+    "28": "2020-11-11",  # Mississippi
+    "34": "2020-09-01",  # New Jersey
+    "38": "2020-11-03",  # North Dakota
+    "46": "2020-11-02",  # South Dakota
+    "47": "2020-09-20",  # Tennessee
+    "53": "2020-10-25",  # Washington
+}
+
+# Mapping of CAN scraper fields to common fields.
+FIELD_MAPPING = {
+    "adult_icu_beds_capacity": CommonFields.ICU_BEDS,
+    "adult_icu_beds_in_use": CommonFields.CURRENT_ICU_TOTAL,
+    "adult_icu_beds_in_use_covid": CommonFields.CURRENT_ICU,
+    "hospital_beds_capacity": CommonFields.STAFFED_BEDS,
+    "hospital_beds_in_use": CommonFields.HOSPITAL_BEDS_IN_USE_ANY,
+    "hospital_beds_in_use_covid": CommonFields.CURRENT_HOSPITALIZED,
+}
+
+
+def make_hhs_variable(can_scraper_field, common_field, measurement):
+    """Helper to create a ScraperVariable, since we have a bunch of variables to deal with and
+    two different measurements to deal with (for county and state data)."""
+    return ccd_helpers.ScraperVariable(
+        variable_name=can_scraper_field,
+        measurement=measurement,
+        provider="hhs",
+        unit="beds",
+        common_field=common_field,
+    )
+
+
+# TODO(michael): HHSHospitalStateDataset and HHSHospitalCountyDataset must be separate classes
+# because they both have ScraperVariables that target the same CommonFields. This happens becaues
+# the state level and county level data use the same variable names but different measurements
+# ("current" and "rolling_average_7_day"). This causes CanScraperBase to overwrite itself as it's
+# extracting the variables. Using separate Dataset classes solves this.
```

Comment:
The problematic code is here: https://github.com/covid-projections/covid-data-model/blob/a4f365631ca0770cfca8a73bcb986f7514521442/libs/datasets/sources/can_scraper_helpers.py#L222

If you think it's worth trying to make it handle this, let me know, though it may introduce some ambiguity (which takes precedence if multiple `ScraperVariable`s that target the same `CommonField` have valid data?).  By treating them as separate datasets, we can use our combined datasets priority merging and there's no ambiguity.  🤷

Reply:
There is a check for multiple elements of VARIABLES with the same `common_field` in the unit test test_can_scraper_class_unique_variable_names. Do you want this to be a run-time assert too?

Reply:
1. Oh, sorry. It's possible I missed that I broke a test since tests were broken in `main` when I was working on this. 😬 
2. As a general rule, yes a runtime assert sounds great.  Anything that can be asserted at runtime should be asserted at runtime, and asserts are better than tests (ideally tests will exercise the asserts and allow you to write less test code that has to be maintained separately).